test/hotspot/jtreg/compiler/vectorapi/TestReverseByteTransforms.java

```diff
@@ -0,0 +1,306 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import java.util.Random;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorSpecies;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.IntVector;
+import jdk.incubator.vector.LongVector;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8287794
+ * @summary Test various reverse bytes ideal transforms.
+ * @requires vm.compiler2.enabled
+ * @modules jdk.incubator.vector
+ * @library /test/lib /
+ * @run driver compiler.vectorapi.TestReverseByteTransforms
+ */
+
+public class TestReverseByteTransforms {
+    static final VectorSpecies<Long> LSPECIES = LongVector.SPECIES_MAX;
+    static final VectorSpecies<Integer> ISPECIES = IntVector.SPECIES_MAX;
+    static final VectorSpecies<Short> SSPECIES = ShortVector.SPECIES_MAX;
+
+    static final int SIZE = 1024;
+    static final int ITERS = 50000;
+
+    static long [] lout = new long[SIZE];
+    static long [] linp = new long[SIZE];
+
+    static int [] iout = new int[SIZE];
+    static int [] iinp = new int[SIZE];
+
+    static short [] sout = new short[SIZE];
+    static short [] sinp = new short[SIZE];
+
+    static void init() {
+        Random r = new Random(1024);
+        for(int i = 0; i < SIZE; i++) {
+            linp[i] = r.nextLong();
+            iinp[i] = r.nextInt();
+            sinp[i] = (short)r.nextInt();
+        }
+    }
+
+    public static void main(String args[]) {
+        init();
+        TestFramework.runWithFlags("-XX:-TieredCompilation", "--add-modules=jdk.incubator.vector");
+        System.out.println("PASSED");
+    }
+
+    @Test
+    @IR(applyIfCPUFeature={"sve", "true"}, failOn = {"ReverseBytesV" , " > 0 "})
+    public void test_reversebytes_long_transform1(long[] lout, long[] linp) {
+        VectorMask<Long> mask = VectorMask.fromLong(LSPECIES, 3);
+        for (int i = 0; i < LSPECIES.loopBound(linp.length); i+=LSPECIES.length()) {
+            LongVector.fromArray(LSPECIES, linp, i)
+                     .lanewise(VectorOperators.REVERSE_BYTES, mask)
+                     .lanewise(VectorOperators.REVERSE_BYTES, mask)
+                     .intoArray(lout, i);
+        }
+    }
+
+    @Run(test = {"test_reversebytes_long_transform1"}, mode = RunMode.STANDALONE)
+    public void kernel_test_reversebytes_long_transform1() {
+        for (int i = 0; i < ITERS; i++) {
+            test_reversebytes_long_transform1(lout, linp);
+        }
+    }
+
+    @Test
+    @IR(applyIfCPUFeatureOr={"sve", "true", "simd", "true", "avx2", "true"}, counts = {"ReverseBytesV" , " > 0 "})
```

Comment:
After https://github.com/openjdk/jdk/pull/9509 merged, I think we'd better to consider different vm flags like "UseAVX", "UseSVE" for each architecture. For example, if the cpu feature mathes "sve", but user may set "-XX:UseSVE=0". With such options, this IR test will also run and I'm afraid it will fail with "-XX:UseSVE=0".

Reply:
CPU features list is populated during VM startup, this list is later ON queried by applyIfCPUFeature during IR validations, I do not think UseSVE is a valid flag for X86, 

```
SPROMPT>java -XX:UseSVE=0
Unrecognized VM option 'UseSVE=0'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
SPROMPT>lscpu
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   46 bits physical, 48 bits virtual
CPU(s):                          72
On-line CPU(s) list:             0-71
Thread(s) per core:              2
Core(s) per socket:              18
Socket(s):                       2
NUMA node(s):                    2
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           85
Model name:                      Intel(R) Xeon(R) Gold 6139 CPU @ 2.30GHz
```

Reply:
Yes, it's invalid on x86. So maybe you could add the limitation to the "requires", but seems this could make the codes complex.

Reply:
Correct.

Reply:
So for such cases, cpu feature "sve" contains "simd", so I think only keep `"simd", "true"` is fine.